docs/proposals/structured-configuration/README.md

```diff
@@ -0,0 +1,543 @@
+---
+title: Structured configuration overrider
+authors:
+- "@Patrick0308"
+- "@sophiefeifeifeiya"
+reviewers:
+- "@chaunceyjiang"
+approvers:
+- "@chaunceyjiang"
+creation-date: 2024-08-12
+---
+
+# Structured configuration overrider
+## Summary
+The proposal introduces a new feature that allows users to partially override values inside  JSON and YAML fields. This is achieved using JSON patch operation. This design enables users to to override the values within JSON/YAML fields partially, rather than replacing a whole JSON/YAML fields with `PlaintextOverrider`. Currently, `PlaintextOverrider`  applies JSON patch operations to whole fields, rather than specific values within fields, making it unsuitable for cases where users need to override individual values within those fields.
```

Comment:
```suggestion
The proposal introduces a new feature that allows users to partially override values inside  JSON and YAML fields. This is achieved using JSON patch operation. This design enables users to override the values within JSON/YAML fields partially, rather than replacing a whole JSON/YAML fields with `PlaintextOverrider`. Currently, `PlaintextOverrider`  applies JSON patch operations to whole fields, rather than specific values within fields, making it unsuitable for cases where users need to override individual values within those fields.
```